src/sqswatcher/plugins/slurm.py

```diff
@@ -208,5 +209,54 @@ def update_cluster(max_cluster_size, cluster_user, update_events, instance_prope
     return failed, succeeded
 
 
+def _is_node_locked(hostname):
+    try:
+        node_state = get_node_state(hostname)
+        log.info("Node %s currently in state %s", hostname, node_state)
+        for disable_state in SLURM_NODE_DISABLED_STATES:
+            if disable_state in node_state:
```

Comment:
same here.
We can have a states like: State=IDLE+DRAIN (see: https://bugs.schedmd.com/show_bug.cgi?id=291) and the disable_state could be a substring of a more complex state.

Let me know if it's by design.

Reply:
This is by design, we only want to check if the locking state is in the node states, and we don't care if the node has other states.